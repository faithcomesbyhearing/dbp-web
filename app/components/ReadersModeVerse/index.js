/**
 *
 * ReadersModeVerse
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import customStyle from '../../utils/customVerseStyle';

function ReadersModeVerse({
	onMouseUp,
	onMouseDown,
	onHighlightClick,
	verse,
	activeVerse,
	verseIsActive,
}) {
  return (
<>
		<span
			onMouseUp={onMouseUp}
			onMouseDown={onMouseDown}
			onClick={onHighlightClick}
			style={customStyle(verse)}
			className={
				verseIsActive &&
				(parseInt(activeVerse, 10) === verse.verse_start ||
					activeVerse === verse.verse_start_alt)
					? 'align-left active-verse'
					: 'align-left'
			}
			data-verseid={verse.verse_start}
			key={verse.verse_start}
			dangerouslySetInnerHTML={{ __html: verse.verse_text }}
		/>
		<span key={`${verse.verse_end}spaces`} className={'readers-spaces'}>
			&nbsp;
		</span>
</>
);
}

ReadersModeVerse.propTypes = {
	onMouseUp: PropTypes.func,
	onMouseDown: PropTypes.func,
	onHighlightClick: PropTypes.func,
	verse: PropTypes.object,
	activeVerse: PropTypes.number,
	verseIsActive: PropTypes.bool,
};

export default ReadersModeVerse;
