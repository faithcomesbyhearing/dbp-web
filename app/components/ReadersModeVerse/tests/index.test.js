import React from 'react';
import Enzyme from 'enzyme';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import renderer from 'react-test-renderer';
import ReadersModeVerse from '..';

Enzyme.configure({ adapter: new Adapter() });

const highlightMessage = 'clicked highlight';

describe('ReadersModeVerse tests', () => {
	let wrapper;
	let onMouseUp = jest.fn();
	let onMouseDown = jest.fn();
	let onHighlightClick = jest.fn();
	let verse = {
		book_id: 'GEN',
		book_name: 'Genesis',
		book_name_alt: 'Genesis',
		chapter: 1,
		chapter_alt: '1',
		verse_end: 1,
		verse_end_alt: '1',
		verse_start: 1,
		verse_start_alt: '1',
		verse_text: 'In the beginning God created the heavens and the earth.',
		wholeVerseHighlighted: true,
	};
	let activeVerse = 0;
	let verseIsActive = false;

	beforeEach(() => {
		onMouseUp = jest.fn();
		onMouseDown = jest.fn();
		onHighlightClick = jest.fn(() => highlightMessage);
		verse = {
			book_id: 'GEN',
			book_name: 'Genesis',
			book_name_alt: 'Genesis',
			chapter: 1,
			chapter_alt: '1',
			verse_end: 1,
			verse_end_alt: '1',
			verse_start: 1,
			verse_start_alt: '1',
			verse_text: 'In the beginning God created the heavens and the earth.',
			wholeVerseHighlighted: true,
		};
		activeVerse = 0;
		verseIsActive = false;
		wrapper = Enzyme.shallow(
			<ReadersModeVerse
				wrapper={wrapper}
				onMouseUp={onMouseUp}
				onMouseDown={onMouseDown}
				onHighlightClick={onHighlightClick}
				verse={verse}
				activeVerse={activeVerse}
				verseIsActive={verseIsActive}
			/>,
		);
	});
	afterEach(() => {
		wrapper = null;
	});
	it('Should match the old snapshot', () => {
		const tree = renderer
			.create(
				<ReadersModeVerse
					wrapper={wrapper}
					onMouseUp={onMouseUp}
					onMouseDown={onMouseDown}
					onHighlightClick={onHighlightClick}
					verse={verse}
					activeVerse={activeVerse}
					verseIsActive={verseIsActive}
				/>,
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('It should shallow render the component and have one verse span', () => {
		expect(wrapper.find('.align-left').length).toEqual(1);
	});
	/* eslint-disable no-underscore-dangle */
	it('It should render the text of the verse passed to it', () => {
		expect(
			wrapper.find('.align-left').props().dangerouslySetInnerHTML.__html,
		).toEqual(verse.verse_text);
	});
	/* eslint-enable no-underscore-dangle */
	it('It should call onHighlightClick when the verse span is clicked', () => {
		const verseNode = wrapper.find('.align-left');
		const spy = jest.spyOn(verseNode.props(), 'onClick');
		const clickResult = verseNode.props().onClick();
		expect(spy).toHaveBeenCalledTimes(1);
		expect(spy).toHaveBeenCalled();
		expect(clickResult).toEqual(highlightMessage);
	});
});
