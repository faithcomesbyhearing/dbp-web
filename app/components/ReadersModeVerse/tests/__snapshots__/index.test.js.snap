// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadersModeVerse tests Should match the snapshot 1`] = `
<DocumentFragment>
  <span
    aria-label="In the beginning God created the heavens and the earth."
    class="align-left"
    data-verseid="1"
  >
    In the beginning God created the heavens and the earth.
  </span>
  <span
    class="readers-spaces"
  >
    Â 
  </span>
</DocumentFragment>
`;
